XenAdmin/Core/Updates.cs

```diff
@@ -120,6 +120,17 @@ public static void RemoveCdnInfoForConnection(IXenConnection connection)
             CdnUpdateInfoChanged?.Invoke(connection);
         }
 
+        public static SyncWithCdnAction CreateSyncWithCdnAction(Pool pool)
+        {
+            var syncAction = new SyncWithCdnAction(pool);
+            syncAction.Completed += a =>
+            {
+                if (a.Succeeded)
+                    CheckForCdnUpdates(a.Connection);
+            };
+            return syncAction;

```

Comment:
Is there an instance whereby we sync and not check for updates? Since we always run `CheckForCdnUpdates` after `SyncWithCdnAction`, should we just be adding `CheckForCdnUpdatesAction` in `SyncWithCdnAction:Run`, or is that a side effect?

Usually I'd be against having one action do two things, but I don't see why we'd run sync without the check afterwards, so it sort of is the same action.

Reply:
The only reason is that `SyncWithCdnAction` has to call `CheckForCdnUpdates`, not `CheckForCdnUpdatesAction`, and the former uses concepts (the Registry) which belong to XenAdmin, not XenModel, and it is not straight forward, if feasible, to move it because it blurrs the interface between business and UI logic (which is already not very clean in some areas).